stdlib/http/src/main/ballerina/src/http/cookie/cookie.bal

```diff
@@ -229,9 +230,18 @@ function parseCookieHeader(string cookieStringValue) returns Cookie[] {
     string cookieValue = cookieStringValue;
     string[] nameValuePairs = stringutils:split(cookieValue, SEMICOLON + SPACE);
     foreach var item in nameValuePairs {
-        string[] nameValue = stringutils:split(item, EQUALS);
-        Cookie cookie = new (nameValue[0], nameValue[1]);
-        cookiesInRequest.push(cookie);
+        if (stringutils:matches(item, "^([^=]+)=.*$")) {
+            string[] nameValue = stringutils:split(item, EQUALS);
+            Cookie cookie;
+            if (nameValue.length() > 1) {
+                cookie = new (nameValue[0], nameValue[1]);
+            } else {
+                cookie = new (nameValue[0], "");
+            }
+            cookiesInRequest.push(cookie);
+        } else {
+            log:printError("Invalid cookie: " + item + ", which must be in the format as [{name}=].");
```

Comment:
Shall we add a test case?

Reply:
Added in the #c6932c65a03c963879c258a151fc0d06b40d957e